modules/storages/lib/api/v3/file_links/work_packages_file_links_api.rb

```diff
@@ -28,50 +28,78 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-class API::V3::FileLinks::WorkPackagesFileLinksAPI < API::OpenProjectAPI
-  # The `:resources` keyword defines the API namespace -> /api/v3/work_packages/:id/file_links/...
-  resources :file_links do
-    get do
-      query = ParamsToQueryService
-                .new(::Storages::Storage,
-                     current_user,
-                     query_class: ::Queries::Storages::FileLinks::FileLinkQuery)
-                .call(params)
+module API
+  module V3
+    module FileLinks
+      class WorkPackagesFileLinksAPI < API::OpenProjectAPI
+        helpers do
+          def sync_and_convert_relation(file_links)
+            return ::Storages::FileLink.none if file_links.empty?
 
-      unless query.valid?
-        message = I18n.t("api_v3.errors.missing_or_malformed_parameter", parameter: "filters")
-        raise ::API::Errors::InvalidQuery.new(message)
-      end
+            sync_result = ::Storages::FileLinkSyncService
+                            .new(user: current_user)
+                            .call(file_links)
+                            .result
```

Comment:
Will it always succeed?

Reply:
The sync service indeed is always successful. While single storage syncs can fail, the sync service only accumulates successes into its own result.